packages/driver/src/cypress/error_utils.ts

```diff
@@ -188,23 +188,28 @@ const appendErrMsg = (err, errMsg) => {
   })
 }
 
-const makeErrFromObj = (obj) => {
+const makeErrFromObj = (obj: any) => {
```

Comment:
Would this type be better?
```suggestion
const makeErrFromObj = (obj?: string | { message: string, name: string, stack?: string }) => {
```

Reply:
`obj?` means the type can be `undefined` **and** the argument can be unfilled.
In our case it must always be filled, even with `undefined`.

Another issue: this typing does not match against objects without `message` or `name` properties. But like `undefined`, it is technically valid to pass any object like `{}` or `{something: 123}` in `throw Error(…)`. So this function should accept anything, and do case by case type handling with a generic fallback.

Maybe [use `unknown` instead of `any`](https://www.typescriptlang.org/docs/handbook/2/functions.html#unknown)?

Reply:
I think in this case we might not be sure what a promise can reject with, so I think `any` is the appropriate type. But we might want to improve the type checking to make sure `message`, `name`, and optional `stack` exist when trying to reconstruct. What do you think?

Reply:
`unknown` is also a good suggestion!

Reply:
Can we really improve the type checking, since we have to accept anything?
Maybe you meant the return type, but that would be `Error`, which provides `name`, `message` and optional `stack` properties in TypeScript:
https://github.com/microsoft/TypeScript/blob/7a38980a7e04bee0e273163f26c91d5c99f28298/src/lib/es5.d.ts#L1057

Reply:
we can improve it slightly where we either expect a string, or an object with the `name`, `message`, and optional `stack` props. If we don't know what it is, we can still throw with the unknown error we have defined since it at least lets end users handle the exception Cypress side.

Reply:
Tried to make this a bit more type safe in b294040b81